zcash_client_sqlite/src/wallet/init/migrations/v_transactions_net.rs

```diff
@@ -0,0 +1,318 @@
+                       CASE
+                           WHEN sent_notes.memo IS NULL THEN 0
+                           ELSE 1
+                       END AS memo_present
+                FROM   transactions
+                       JOIN sent_notes ON transactions.id_tx = sent_notes.tx
+            ) AS notes
+            LEFT JOIN blocks ON notes.mined_height = blocks.height
+            GROUP BY notes.id_tx",
+        )?;
+
+        Ok(())
+    }
+
+    fn down(&self, _transaction: &rusqlite::Transaction) -> Result<(), WalletMigrationError> {
+        // TODO: something better than just panic?
+        panic!("Cannot revert this migration.");
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use rusqlite::{self, params};
+    use tempfile::NamedTempFile;
+
+    use zcash_client_backend::keys::UnifiedSpendingKey;
+    use zcash_primitives::zip32::AccountId;
+
+    use crate::{
+        tests,
+        wallet::init::{
+            init_wallet_db, init_wallet_db_internal, migrations::add_transaction_views,
+        },
+        WalletDb,
+    };
+
+    #[test]
+    fn v_transactions_net() {
+        let data_file = NamedTempFile::new().unwrap();
+        let mut db_data = WalletDb::for_path(data_file.path(), tests::network()).unwrap();
+        init_wallet_db_internal(&mut db_data, None, &[add_transaction_views::MIGRATION_ID])
+            .unwrap();
+        let usk =
+            UnifiedSpendingKey::from_seed(&tests::network(), &[0u8; 32][..], AccountId::from(0))
+                .unwrap();
+        let ufvk = usk.to_unified_full_viewing_key();
+
+        db_data
+            .conn
+            .execute(
+                "INSERT INTO accounts (account, ufvk) VALUES (0, ?)",
+                params![ufvk.encode(&tests::network())],
+            )
+            .unwrap();
+
+        db_data.conn.execute_batch(
+            "INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (0, 0, 0, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (0, 0, 'a');
+
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 0, 0, '', 2, '', 'a', false);
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (0, 3, 0, '', 5, '', 'b', false);
+
+            INSERT INTO blocks (height, hash, time, sapling_tree) VALUES (1, 1, 1, '');
+            INSERT INTO transactions (block, id_tx, txid) VALUES (1, 1, 'b');
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_address, value)
+            VALUES (1, 2, 0, 0, '', 2);
+            INSERT INTO sent_notes (tx, output_pool, output_index, from_account, to_address, value, memo)
+            VALUES (1, 2, 1, 0, '', 3, X'61');
+            INSERT INTO received_notes (tx, output_index, account, diversifier, value, rcm, nf, is_change)
+            VALUES (1, 3, 0, '', 2, '', 'c', true);",
```

Comment:
Add a check that after the new migration is run, the only change to the data in the database is that `sent_notes` contains a row corresponding to this change note in `received_notes`.